backend/src/main/java/wooteco/retrospective/config/AuthenticationPrincipalConfig.java

```diff
@@ -0,0 +1,39 @@
+package wooteco.retrospective.config;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+import wooteco.retrospective.application.member.MemberService;
+import wooteco.retrospective.infrastructure.auth.JwtTokenProvider;
+import wooteco.retrospective.presentation.auth.AuthenticationPrincipalArgumentResolver;
+
+import java.util.List;
+
+@Configuration
+public class AuthenticationPrincipalConfig implements WebMvcConfigurer {
+
+    private final MemberService memberService;
+    private final JwtTokenProvider jwtTokenProvider;
+
+    public AuthenticationPrincipalConfig(MemberService memberService, JwtTokenProvider jwtTokenProvider) {
+        this.memberService = memberService;
+        this.jwtTokenProvider = jwtTokenProvider;
+    }
+
+    @Override
+    public void addArgumentResolvers(List argumentResolvers) {
+        argumentResolvers.add(createAuthenticationPrincipalArgumentResolver());
+    }
+
+    @Bean
+    public AuthenticationPrincipalArgumentResolver createAuthenticationPrincipalArgumentResolver() {
+        return new AuthenticationPrincipalArgumentResolver(memberService);
+    }
+
+    @Override
+    public void addInterceptors(InterceptorRegistry registry) {
+        registry.addInterceptor(new LoginInterceptor(jwtTokenProvider))
+                .excludePathPatterns("/api/login", "/index.html","/");
```

Comment:
현재 우리의 프론트-백엔드 기술 스택은,  SSR이 아니기 때문에 프론트 서버가 따로 동작하는 형태로 될꺼에요. 
아마 구성되는 방식은, 도메인이 있다면 A레코드를 이용해서 API서버와 프론트 서버를 도메인 상으로 분리시키겠죠.
도메인은 동일한 웹 서버와 연결되어 있고, 이게 리버스 프록시 서버가 될꺼에요. 그러면 리버스 프록시에서 가상 호스트 기술을 이용해서 도메인별로 요청을 분리시키게 될 겁니다. API서버로 가는 요청은 API로 보낼꺼고, 프론트로 오는 요청은 react서버로 요청하게 될꺼에요. 리버스 프록시를 사용하지 않더라도 부트에서 제공하는 기본 웹 서버에서 리액트와 부트로의 요청을 분리하게 될겁니다.

이 말을 왜 했느냐. 인터셉터에서 굳이 index.html의 exclude로직을 안해줘도 된다는 의미에요. 우리가 SSR을 사용하는것도 아니고 부트의 static resource에 접근하는 방식도 아니기 때문에 해당 부분이 필요가 없다고 생각됩니다

```suggestion
                .excludePathPatterns("/api/login");
```

이렇게 하더라도 동작에 무리가 없을꺼에요.

Reply:
첫 장을 index.html로 뽑는 거 아닌가요? 소롱 등판해주세요